lib/Models/addUserCatalogMember.ts

```diff
@@ -0,0 +1,71 @@
+"use strict";
+
+import Terria from "./Terria";
+import { BaseModel } from "./Model";
+import isDefined from "../Core/isDefined";
+import hasTraits from "./hasTraits";
+import GroupTraits from "../Traits/GroupTraits";
+import when from "terriajs-cesium/Source/ThirdParty/when";
+import TerriaError from "../Core/TerriaError";
+import CommonStrata from "./CommonStrata";
+import Mappable from "./Mappable";
+import defaultValue from "terriajs-cesium/Source/Core/defaultValue";
+import GeoJsonCatalogItem from "./GeoJsonCatalogItem";
+import GroupMixin from "../ModelMixins/GroupMixin";
+
+interface AddUserCatalogMemberOptions {
+    enable?: boolean;
+    open?: boolean;
+    zoomTo?: boolean;
+}
+
+/**
+ * Adds a user's catalog item or group to the catalog.
+ *
+ */
+export default function addUserCatalogMember(
+    terria: Terria,
+    newCatalogMemberOrPromise: BaseModel | Promise<BaseModel | undefined>,
+    optionsArg?: AddUserCatalogMemberOptions
+) {
+    const options: AddUserCatalogMemberOptions = defaultValue(optionsArg, {});
```

Comment:
It's ok to just call `optionsArg` `options` and then reassign it with the default value:
```
options = defaultValue(options, {});
```

Sometimes considered a bit dodgy to reassign a parameter like this, but it's IMO ok in this case and it's very commonly done in Cesium and TerriaJS.

Reply:
I guess TypeScript might fight you on this though eh? hmm

Reply:
Yeah, TS gives an error because the references are inside a closure. Can fix it though, will have to do an `isDefined(options)` test.